app/forms/PassportOrIdCardFormProvider.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import java.time.LocalDate
+
+import javax.inject.Inject
+import forms.mappings.Mappings
+import play.api.data.Form
+import play.api.data.Forms._
+import models.PassportOrIdCardDetails
+
+
+class PassportOrIdCardFormProvider @Inject() extends Mappings {
+
+
+  def apply(): Form[PassportOrIdCardDetails] = Form(
+    mapping(
+      "country" -> text("settlorIndividualPassport.country.error.required")
+        .verifying(
+          firstError(
+            maxLength(100, "settlorIndividualPassport.country.error.length"),
+            isNotEmpty("country", "settlorIndividualPassport.country.error.required")
+          )
+        ),
+      "number" -> text("settlorIndividualPassport.number.error.required")
+        .verifying(
+          firstError(
+            maxLength(30, "settlorIndividualPassport.number.error.length"),
+            regexp(Validation.passportOrIdCardNumberRegEx, "settlorIndividualPassport.number.error.invalid"),
+            isNotEmpty("number", "settlorIndividualPassport.number.error.required")
+          )
+        ),
+      "expiryDate" -> localDate(
+        invalidKey     = "settlorIndividualPassport.expiryDate.error.invalid",
+        allRequiredKey = "settlorIndividualPassport.expiryDate.error.required.all",
+        twoRequiredKey = "settlorIndividualPassport.expiryDate.error.required.two",
+        requiredKey    = "settlorIndividualPassport.expiryDate.error.required"
+      ).verifying(firstError(
+        maxDate(LocalDate.now, s"settlorIndividualPassport.expiryDate.error.future", "day", "month", "year"),
```

Comment:
ID Card can have a future expiry date, this only allows up to today.
Error messages should relate to ID Card and not Passport.

Reply:
This has now been set to 2100 as the maximum expiry date.